akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpResponseParser.scala

```diff
@@ -70,7 +71,17 @@ private[http] class HttpResponseParser(protected val settings: ParserSettings, p
         case 200 => StatusCodes.OK
         case code => StatusCodes.getForKey(code) match {
           case Some(x) => x
-          case None    => customStatusCodes(code) getOrElse badStatusCodeSpecific(code)
+          case None => customStatusCodes(code) getOrElse {
+            // A client must understand the class of any status code, as indicated by the first digit, and
+            // treat an unrecognized status code as being equivalent to the x00 status code of that class
+            // https://tools.ietf.org/html/rfc7231#section-6
+            try {
+              val reason = asciiString(input, reasonStartIdx, reasonEndIdx)
+              StatusCodes.custom(code, reason)
+            } catch {
+              case _: Exception => badStatusCodeSpecific(code)
```

Comment:
should this not be `case NonFatal(_) =>`?

Reply:
Thanks for the extra set of eyes! That indeed seems reasonable, would you be interested in doing a follow-up PR?